src/engine/loopingcontrol.cpp

```diff
@@ -231,76 +298,43 @@ void LoopingControl::slotLoopScale(double scale) {
     }
 
     m_loopSamples.setValue(loopSamples);
-  
+
     // Update CO for loop end marker
     m_pCOLoopEndPosition->set(loopSamples.end);
 
     // Reseek if the loop shrank out from under the playposition.
-    if (m_bLoopingEnabled && scale < 1.0) {
+    if (m_bLoopingEnabled && scaleFactor < 1.0) {
         seekInsideAdjustedLoop(
                 loopSamples.start, old_loop_end,
                 loopSamples.start, loopSamples.end);
     }
 }
 
-void LoopingControl::slotLoopHalve(double v) {
-    LoopSamples loopSamples = m_loopSamples.getValue();
-    if (loopSamples.start == kNoTrigger || 
-            loopSamples.end == kNoTrigger ||
-            v <= 0.0) {
+void LoopingControl::slotLoopHalve(double pressed) {
+    if (pressed <= 0.0) {
         return;
     }
 
-    // If a beatloop is active then halve should deactive the current
-    // beatloop and activate the previous one.
-    BeatLoopingControl* pActiveBeatLoop = m_pActiveBeatLoop;
-    if (pActiveBeatLoop != nullptr) {
-        int active_index = m_beatLoops.indexOf(pActiveBeatLoop);
-        if (active_index - 1 >= 0) {
-            if (m_bLoopingEnabled) {
-                // If the current position is outside the range of the new loop,
-                // take the current position and subtract the length of the new loop until
-                // it fits.
-                int old_loop_in = loopSamples.start;
-                int old_loop_out = loopSamples.end;
-                slotBeatLoopActivate(m_beatLoops[active_index - 1]);
-                loopSamples = m_loopSamples.getValue();
-                seekInsideAdjustedLoop(
-                        old_loop_in, old_loop_out,
-                        loopSamples.start, loopSamples.end);
-            } else {
-                // Calling scale clears the active beatloop.
-                slotLoopScale(0.5);
-                m_pActiveBeatLoop = m_beatLoops[active_index - 1];
-            }
-        }
+    LoopSamples loopSamples = m_loopSamples.getValue();
+    bool noLoopSet = loopSamples.start == kNoTrigger || loopSamples.end == kNoTrigger;
+
+    if (noLoopSet || currentLoopMatchesBeatloopSize()) {
+        slotBeatLoop(m_pCOBeatLoopSize->get() / 2.0, true, false);
```

Comment:
Why is this not done for "loop_scale"? This means "loop_scale" will exit an active beat loop. This is probably not intended?